src/state.ts

```diff
@@ -26,32 +32,59 @@ interface ViewState {
   default?: boolean;
 }
 
-interface BaseConfig {
-  source: string | ZarrArray['store'];
-  axis_labels?: string[];
-  name?: string;
-  colormap?: string;
-  opacity?: number;
-  acquisition?: string;
-  model_matrix?: string | number[];
-  onClick?: (e: any) => void;
-}
-
-export interface MultichannelConfig extends BaseConfig {
-  colors?: string[];
-  channel_axis?: number;
-  contrast_limits?: number[][];
-  names?: string[];
-  visibilities?: boolean[];
-}
+const isStore = (u: any): u is ZarrArray['store'] => {
+  const isObject = typeof u === 'object' && u !== null;
+  return isObject && u.getItem === 'function' && u.containsItem === 'function';
+};
 
-export interface SingleChannelConfig extends BaseConfig {
-  color?: string;
-  contrast_limits?: number[];
-  visibility?: boolean;
-}
+const ZarrStore = new t.Type<ZarrArray['store'], ZarrArray['store'], unknown>(
+  'ZarrStore',
+  isStore,
+  (input, context) => (isStore(input) ? t.success(input) : t.failure(input, context)),
+  t.identity
+);
```

Comment:
If/when we merge #116, we can move `normalizeStore` to this type.

```typescript
ZarrStore.decode("https://localhost:8080/data.zarr"); // Either<Errors, HTTPStore>
ZarrStore.decode(new CustomStore(...)); // Either<Errors, CustomStore>
```

Reply:
My hope (ideally) is to avoid extra code to handle edge cases within `src/ome.ts` and `src/io.ts`. Instead, we can perform validation and coerce types within `src/vizarr.ts` via `ImageLayerConfig.decode(unknownConfig)` so that the rest of the code in our system doesn't need to try to account for the different outer APIs.